lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -1451,6 +1451,8 @@ namespace Js
 
     RecyclableObject* JavascriptProxy::GetPrototypeSpecial()
     {
+        PROBE_STACK(GetScriptContext(), Js::Constants::MinStackDefault);
```

Comment:
SetProperty calls GetOwnPropertyDescriptor, which eventually has it. but I don't have test case to back that up. I can add the check there as well. 
I didn't understand the IsExtensible question.